### ISSUE TITLE
Bluetooth connection drops

### ISSUE THREAD

**Issue by McRender** [fhorinek/SkyDrop#268]
Hi there,

I'am using the SkyDrop (rev 1506, build 3395) in combination with a Samsung Galaxy Tab A6 7'' and the XCSoar software. My communication runs over the Digifly protocol to get vario, gps and baro data. As long as there is no GPS fix on the SkyDrop everything works fine. But when there is a GPS fix the BT connection fails after about 20' - 60'. When scrolling through the debug.log file there were several entries with "bt output buffer full! Am I reading too slow on my Android device or do you have any idea how to keep the connection running? 

Would love to here from you. Greetings from Switzerland,
Stephan

**Comment by fhorinek** [comment]
it looks like bt module does not have enough time to transmit the data.
Try to upgrade to latest version 3466
what Bluetooth module do you have Pan1322 or pan1026?

**Comment by McRender** [comment]
Thank you for the quick response. I'll give it a try with the new version 3466. The module is a pan1026 v4.0 bought about three month ago.

**Comment by fhorinek** [comment]
Also try it without data files for agl. It is new function and might cause
some hidden problems

**Comment by McRender** [comment]
Run some test with the new firmware. When there is no GPS fix everything run as expected. After getting the fix I was standing at the same position for about 5' without any problems. After heading in a direction by foot no more data was transmitted over BT.

I'll give it a try without the AGL files now...

[DEBUG.txt](https://github.com/fhorinek/SkyDrop/files/851649/DEBUG.txt)

**Comment by McRender** [comment]
Ok... just deleted the agl files and gave it another try. Problem remains as described in the first post.

[DEBUG.txt](https://github.com/fhorinek/SkyDrop/files/851667/DEBUG.txt)

**Comment by McRender** [comment]
Allright... next test was with the XCTrack App and I ran into the same issue.

**Comment by McRender** [comment]
Hi fhorinek, do you have any updates or suggestions on this issue? Greetings from Switzerland.

**Comment by fhorinek** [comment]
it is highly probable that the problem is in skydrop. I will need to do some testing but unfortunately I have very little time to do that.
Does this problem occurred before?

**Comment by McRender** [comment]
I first noticed this issue when I was connecting my Skydrop via bluetooth a few days ago. Bought the Skydrop about three month ago in Switzerland and used it as a standlone device. But the feature of connecting it to an Android tablet is kindly welcome to me :-). Let me know if I can do anything for you, since I'am a software engineer I might be able to run some test or play with the code.

**Comment by fhorinek** [comment]
what is in xcsoar monitor tab? any malformed messages?

**Comment by McRender** [comment]
I'am not sure if I do configure xcsoar correct for the use with my skydrop. I have attached all the log files from xcsoar and skydrop as well as a screenshot from the monitor tab and my settings. Let me know if you need anything else. 

[xcsoar_monitor.txt](https://github.com/fhorinek/SkyDrop/files/874855/xcsoar_monitor.txt)
[xcsoar_log.txt](https://github.com/fhorinek/SkyDrop/files/874857/xcsoar_log.txt)
![screenshot_2017-03-28-09-48-00](https://cloud.githubusercontent.com/assets/5795337/24394640/0667eec8-139d-11e7-9c2c-192c01c9932e.png)
![screenshot_2017-03-28-09-48-34](https://cloud.githubusercontent.com/assets/5795337/24394644/08c1fbaa-139d-11e7-90e0-826532f81148.png)
[DEBUG.txt](https://github.com/fhorinek/SkyDrop/files/874858/DEBUG.txt)

**Comment by mserralta** [comment]
Hi there!

Yesterday I lost my Bluetooth connection as well. I don't know if it might help or not... I just comment.

I realized I can pair my phone and Skydrop with two differents Bluetooth interfaces. Bluetooth SPP and Bluetooth BLE. I don't know exactly the diference but yesterday I tried BLE (I guessed, maybe wrongly, it consumes less energy). Anyway... I lost my Bluetooth connection using this BLE interfaces. Then I changed to SPP as I was used to and all work fine.

Am I supposed to use always SPP? Is it more trustable?

**Comment by McRender** [comment]
Hi mserralta,

thanks for sharing your experience with us. I have also tried both connection methodes (SPP and BLE) but the result was always the same. Do you have a debug.log to share? Would be interesting to see if you also have the "bt output buffer full!" warning.

**Comment by mserralta** [comment]
Do I have to activate any option to log the file?? I did not. If the
debug.log is created automatically inside the SkyDrop I can send you this
evening

2017-03-30 9:56 GMT+02:00 McRender <notifications@github.com>:

**Comment by McRender** [comment]
Yeah I think you have to activate the Debug log option under Settings/Debug. If you don't mind just activate it and see if the connection works the next time you use it. Otherwise we will have another log file to play with.

**Comment by mserralta** [comment]
I will try with BLE and log activated next time and, if occurs, I let you know!

**Comment by mserralta** [comment]
It occured again but in LOG folder I just have the IGC file. Which other file is supposed to be? I flew with debug mode on and I lost bluetooth connection with my phone during the flight. I was not able to restart the connection.

Always using BLE. This weekend I will switch to SPP again (still don't know the difference, indeed)

**Comment by McRender** [comment]
If you activate the debug mode there should be a file caled DEBUG.LOG in the root directory of your SDCard:

![folder](https://cloud.githubusercontent.com/assets/5795337/24791433/5b6f98b4-1b7a-11e7-85da-944f14fc3665.jpg)

**Comment by mserralta** [comment]
It was not...

Should this file exists only in case of crash or error in application or is
it there just when you start in debug mode? Maybe there was not error in
SkyDrop. I realized I was flying in mobile phone "Eco mode" turned on.
Maybe the long time bluetooth connection was dropped off because of this
mode.

I will try again this weekend...

Cheers

2017-04-07 10:10 GMT+02:00 McRender <notifications@github.com>:

**Comment by McRender** [comment]
Make sure you've got the latest firmware on your device and than go to Options-> Debug and set the debug log tick.

![Uploading Unbenannt-1.jpg…]()

**Comment by fhorinek** [comment]
BLE means bluetooth low energy

**Comment by st0rmch4s3r** [comment]
Same problem still existing.
Using build 5265 and rev 1506 coupled with android.
Everything works fine till a GPS fix happens, after that the data transmission stops.
Xc Track continues telling it's coupled with the BT device but won't receive data anymore.
Any solution coming?
Thanks

**Comment by mkflying** [comment]
Same problem here with the skydrop built 5265 rev 1506.
Please I need a soltion. Was one of the main reason why I have baught it...
Thanks in advance!

**Comment by mkflying** [comment]
I connected it via BLE now.
And disabled every "available data" input in xcTrack except for "GPS" and "Barometric pressure".
Now it works as it should.
hopefully it helps somebody.

BTW: at first for some reason connecting via BLE did not work. Then I have disabled "foreward GPS" in Bluetooth settings on my Skydrop. Afterwards connecting via BLE worked just fine. Then I enabled "foreward GPS" on my skydrop again and no problems with any disconnections to xcTracker as sensor anymore.

**Comment by mkflying** [comment]
After disabling bluetooth on my skydrop and re-enabling without restarting, I had the connection issue again.
So alway restart after changing something.
